### ISSUE TITLE
distutils is deprecated and removed in Python 3.12

### ISSUE THREAD

**Issue by szaszlyki21** [dipu-bd/lightnovel-crawler#2368]
## Describe the bug

I'm using an arch distro which uses the newest stable release of python which is 3.12. With 3.12 distutils is now deprecated, so the program errors out for not finding distutils in my isolated Virtual Environment.

logs:
```
Module load failed: /home/user/.local/pipx/venvs/lightnovel-crawler/lib/python3.12/site-packages/sources/multi/novelupdates.py | No module named 'distutils'
Module load failed: /home/user/.local/pipx/venvs/lightnovel-crawler/lib/python3.12/site-packages/sources/fr/lightnovelfr.py | No module named 'distutils'
Module load failed: /home/user/.local/pipx/venvs/lightnovel-crawler/lib/python3.12/site-packages/sources/pt/centralnovel.py | No module named 'distutils'
Module load failed: /home/user/.local/pipx/venvs/lightnovel-crawler/lib/python3.12/site-packages/sources/ar/kolnovel.py | No module named 'distutils'
Module load failed: /home/user/.local/pipx/venvs/lightnovel-crawler/lib/python3.12/site-packages/sources/ar/rewayatclub.py | 3
Module load failed: /home/user/.local/pipx/venvs/lightnovel-crawler/lib/python3.12/site-packages/sources/en/t/tamagotl.py | No module named 'distutils'
Module load failed: /home/user/.local/pipx/venvs/lightnovel-crawler/lib/python3.12/site-packages/sources/en/p/pandamanga.py | No module named 'distutils'
Module load failed: /home/user/.local/pipx/venvs/lightnovel-crawler/lib/python3.12/site-packages/sources/en/w/wuxiacom.py | No module named 'distutils'
Module load failed: /home/user/.local/pipx/venvs/lightnovel-crawler/lib/python3.12/site-packages/sources/en/w/webnovel.py | No module named 'distutils'
Module load failed: /home/user/.local/pipx/venvs/lightnovel-crawler/lib/python3.12/site-packages/sources/en/w/webnovelpub.py | No module named 'distutils'
Module load failed: /home/user/.local/pipx/venvs/lightnovel-crawler/lib/python3.12/site-packages/sources/en/l/lnmtl.py | 3
Module load failed: /home/user/.local/pipx/venvs/lightnovel-crawler/lib/python3.12/site-packages/sources/en/l/lightnovelpub.py | No module named 'distutils'
Module load failed: /home/user/.local/pipx/venvs/lightnovel-crawler/lib/python3.12/site-packages/sources/en/l/lightnovelworld.com.py | No module named 'distutils'
Module load failed: /home/user/.local/pipx/venvs/lightnovel-crawler/lib/python3.12/site-packages/sources/en/r/ranobes.py | 3
Module load failed: /home/user/.local/pipx/venvs/lightnovel-crawler/lib/python3.12/site-packages/sources/en/s/scribblehub.py | No module named 'distutils'
Module load failed: /home/user/.local/pipx/venvs/lightnovel-crawler/lib/python3.12/site-packages/sources/en/s/systemtranslation.py | No module named 'distutils'
Module load failed: /home/user/.local/pipx/venvs/lightnovel-crawler/lib/python3.12/site-packages/sources/en/m/mltnovels.py | No module named 'distutils'
Module load failed: /home/user/.local/pipx/venvs/lightnovel-crawler/lib/python3.12/site-packages/sources/en/i/imperfectcomic.py | No module named 'distutils'
Module load failed: /home/user/.local/pipx/venvs/lightnovel-crawler/lib/python3.12/site-packages/sources/en/n/novelsonline.py | No module named 'distutils'
Module load failed: /home/user/.local/pipx/venvs/lightnovel-crawler/lib/python3.12/site-packages/sources/en/n/noblemtl.py | No module named 'distutils'
Module load failed: /home/user/.local/pipx/venvs/lightnovel-crawler/lib/python3.12/site-packages/sources/en/n/novelpub.py | No module named 'distutils'
Module load failed: /home/user/.local/pipx/venvs/lightnovel-crawler/lib/python3.12/site-packages/sources/en/n/novelmao.py | 3
Module load failed: /home/user/.lncrawl/sources/en/n/novelsonline.py | No module named 'distutils'
Module load failed: /home/user/.lncrawl/sources/en/r/ranobes.py | 3
Module load failed: /home/user/.lncrawl/sources/en/s/scribblehub.py | No module named 'distutils'
Module load failed: /home/user/.lncrawl/sources/en/w/wuxiacom.py | No module named 'distutils'
Module load failed: /home/user/.lncrawl/sources/ar/kolnovel.py | No module named 'distutils'

 ❗ Error: No module named 'distutils' 
<class 'ModuleNotFoundError'>
File "/home/user/.local/pipx/venvs/lightnovel-crawler/lib/python3.12/site-packages/lncrawl/webdriver/__init__.py", line 12, in <module>
    from .local import create_local
  File "/home/user/.local/pipx/venvs/lightnovel-crawler/lib/python3.12/site-packages/lncrawl/webdriver/local.py", line 12, in <module>
    from undetected_chromedriver import Chrome, ChromeOptions
  File "/home/user/.local/pipx/venvs/lightnovel-crawler/lib/python3.12/site-packages/undetected_chromedriver/__init__.py", line 44, in <module>
    from .patcher import IS_POSIX
  File "/home/user/.local/pipx/venvs/lightnovel-crawler/lib/python3.12/site-packages/undetected_chromedriver/patcher.py", line 4, in <module>
    from distutils.version import LooseVersion

```
**App source**:    PIP 
**App version**:   v3.6.0
**Your OS**:       Linux

**Comment by sourajit02** [comment]
as a temporary workaround for those using pipx, install python3.11 and then install lncrawl as follows:
`pipx install lightnovel-crawler --python /usr/bin/python3.11`

**Comment by sourajit02** [comment]
@korason7117 then instead of downgrading, maybe try specifying setuptools (which contains distutils) as an explicit dependancy: `pipx install lightnovel-crawler --pip-args="setuptools"`
Both methods work for me, also on an uptodate arch system btw.

**Comment by korason7117** [comment]
@sourajit02 it works. only thing i see are these `Module load failled` messages
```
Module load failed: /home/user/.local/pipx/venvs/lightnovel-crawler/lib/python3.12/site-packages/sources/ar/rewayatclub.py | 3 
Module load failed: /home/user/.local/pipx/venvs/lightnovel-crawler/lib/python3.12/site-packages/sources/en/n/novelmao.py | 3 
Module load failed: /home/user/.local/pipx/venvs/lightnovel-crawler/lib/python3.12/site-packages/sources/en/r/ranobes.py | 3 
Module load failed: /home/user/.local/pipx/venvs/lightnovel-crawler/lib/python3.12/site-packages/sources/en/l/lnmtl.py | 3

```

**Comment by RayZ3R0** [comment]
if you use pyenv you can also do this

```
pipx install --python $(pyenv prefix 3.11.10)/bin/python lightnovel-crawler
```

**Comment by dipu-bd** [comment]
should be fixed in v3.7.4

**Comment by korason7117** [comment]
> should be fixed in v3.7.4

nah.

```
lncrawl https://novelbin.me/novel-book/global-game-afk-in-the-zombie-apocalypse-game                                                                                       
================================================================================
╭╮╱╱╱╱╱╱╭╮╱╭╮╱╱╱╱╱╱╱╱╱╱╱╱╭╮╱╭━━━╮╱╱╱╱╱╱╱╱╱╭╮
┃┃╱╱╱╱╱╱┃┃╭╯╰╮╱╱╱╱╱╱╱╱╱╱╱┃┃╱┃╭━╮┃╱╱╱╱╱╱╱╱╱┃┃
┃┃╱╱╭┳━━┫╰┻╮╭╋━╮╭━━┳╮╭┳━━┫┃╱┃┃╱╰╋━┳━━┳╮╭╮╭┫┃╭━━┳━╮
┃┃╱╭╋┫╭╮┃╭╮┃┃┃╭╮┫╭╮┃╰╯┃┃━┫┃╱┃┃╱╭┫╭┫╭╮┃╰╯╰╯┃┃┃┃━┫╭╯
┃╰━╯┃┃╰╯┃┃┃┃╰┫┃┃┃╰╯┣╮╭┫┃━┫╰╮┃╰━╯┃┃┃╭╮┣╮╭╮╭┫╰┫┃━┫┃
╰━━━┻┻━╮┣╯╰┻━┻╯╰┻━━╯╰╯╰━━┻━╯╰━━━┻╯╰╯╰╯╰╯╰╯╰━┻━━┻╯
╱╱╱╱╱╭━╯┃ v3.7.4
╱╱╱╱╱╰━━╯ 🔗 https://github.com/dipu-bd/lightnovel-crawler
--------------------------------------------------------------------------------
Sources: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:00<00:00,  4.01file/s]
Module load failed: /home/user/.local/pipx/venvs/lightnovel-crawler/lib/python3.12/site-packages/sources/multi/novelupdates.py | No module named 'distutils'
Module load failed: /home/user/.local/pipx/venvs/lightnovel-crawler/lib/python3.12/site-packages/sources/fr/lightnovelfr.py | No module named 'distutils'
Module load failed: /home/user/.local/pipx/venvs/lightnovel-crawler/lib/python3.12/site-packages/sources/ar/kolnovel.py | No module named 'distutils'
Module load failed: /home/user/.local/pipx/venvs/lightnovel-crawler/lib/python3.12/site-packages/sources/ar/rewayatclub.py | 3
Module load failed: /home/user/.local/pipx/venvs/lightnovel-crawler/lib/python3.12/site-packages/sources/pt/centralnovel.py | No module named 'distutils'
Module load failed: /home/user/.local/pipx/venvs/lightnovel-crawler/lib/python3.12/site-packages/sources/en/p/pandamanga.py | No module named 'distutils'
Module load failed: /home/user/.local/pipx/venvs/lightnovel-crawler/lib/python3.12/site-packages/sources/en/m/mltnovels.py | No module named 'distutils'
Module load failed: /home/user/.local/pipx/venvs/lightnovel-crawler/lib/python3.12/site-packages/sources/en/n/novelpub.py | No module named 'distutils'
Module load failed: /home/user/.local/pipx/venvs/lightnovel-crawler/lib/python3.12/site-packages/sources/en/n/noblemtl.py | No module named 'distutils'
Module load failed: /home/user/.local/pipx/venvs/lightnovel-crawler/lib/python3.12/site-packages/sources/en/n/novelmao.py | 3
Module load failed: /home/user/.local/pipx/venvs/lightnovel-crawler/lib/python3.12/site-packages/sources/en/n/novelsonline.py | No module named 'distutils'
Module load failed: /home/user/.local/pipx/venvs/lightnovel-crawler/lib/python3.12/site-packages/sources/en/e/ebotnovel.py | No module named 'distutils'
Module load failed: /home/user/.local/pipx/venvs/lightnovel-crawler/lib/python3.12/site-packages/sources/en/r/ranobes.py | 3
Module load failed: /home/user/.local/pipx/venvs/lightnovel-crawler/lib/python3.12/site-packages/sources/en/t/tamagotl.py | No module named 'distutils'
Module load failed: /home/user/.local/pipx/venvs/lightnovel-crawler/lib/python3.12/site-packages/sources/en/c/ckandawrites.online.py | No module named 'distutils'
Module load failed: /home/user/.local/pipx/venvs/lightnovel-crawler/lib/python3.12/site-packages/sources/en/l/lnmtl.py | 3
Module load failed: /home/user/.local/pipx/venvs/lightnovel-crawler/lib/python3.12/site-packages/sources/en/l/lightnovelworld.com.py | No module named 'distutils'
Module load failed: /home/user/.local/pipx/venvs/lightnovel-crawler/lib/python3.12/site-packages/sources/en/l/lightnovelpub.py | No module named 'distutils'
Module load failed: /home/user/.local/pipx/venvs/lightnovel-crawler/lib/python3.12/site-packages/sources/en/w/webnovelpub.py | No module named 'distutils'
Module load failed: /home/user/.local/pipx/venvs/lightnovel-crawler/lib/python3.12/site-packages/sources/en/w/wuxiacom.py | No module named 'distutils'
Module load failed: /home/user/.local/pipx/venvs/lightnovel-crawler/lib/python3.12/site-packages/sources/en/w/webnovel.py | No module named 'distutils'
Module load failed: /home/user/.local/pipx/venvs/lightnovel-crawler/lib/python3.12/site-packages/sources/en/s/scribblehub.py | No module named 'distutils'
Module load failed: /home/user/.local/pipx/venvs/lightnovel-crawler/lib/python3.12/site-packages/sources/en/s/systemtranslation.py | No module named 'distutils'
Module load failed: /home/user/.local/pipx/venvs/lightnovel-crawler/lib/python3.12/site-packages/sources/en/i/imperfectcomic.py | No module named 'distutils'
Module load failed: /home/user/.lncrawl/sources/pt/centralnovel.py | No module named 'distutils'
Module load failed: /home/user/.lncrawl/sources/en/e/ebotnovel.py | No module named 'distutils'
Module load failed: /home/user/.lncrawl/sources/en/c/ckandawrites.online.py | No module named 'distutils'

➡ Press  Ctrl + C  to exit


 ❗ Error: No module named 'distutils' 
<class 'ModuleNotFoundError'>
File "/home/user/.local/pipx/venvs/lightnovel-crawler/lib/python3.12/site-packages/lncrawl/webdriver/__init__.py", line 12, in <module>
    from .local import create_local
  File "/home/user/.local/pipx/venvs/lightnovel-crawler/lib/python3.12/site-packages/lncrawl/webdriver/local.py", line 12, in <module>
    from undetected_chromedriver import Chrome, ChromeOptions
  File "/home/user/.local/pipx/venvs/lightnovel-crawler/lib/python3.12/site-packages/undetected_chromedriver/__init__.py", line 44, in <module>
    from .patcher import IS_POSIX
  File "/home/user/.local/pipx/venvs/lightnovel-crawler/lib/python3.12/site-packages/undetected_chromedriver/patcher.py", line 4, in <module>
    from distutils.version import LooseVersion


--------------------------------------------------------------------------------
 🔗  https://github.com/dipu-bd/lightnovel-crawler/issues 
================================================================================
```
